### PR TITLE
Add clang-format-diff hook

### PR DESCRIPTION
**Pull request by aheejin**
This adds a commit hook to Travis CI that errors out if incoming PRs'
diffs are not clang-formatted. Turns out clang-format is also capable of
formatting JavaScript, but we haven't agreed on a style for JS yet, this
PR disables JavaScript formatting for now. This also adds clang-format
exempt header/footer to a generated source file.

### CODE REVIEW

File: test/bad_params.asm.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ function asm(global, env, buffer) {
   "use asm";
 
   function missing(x, y) {
-    x = x | 0;
+    x = x | 0; a = 3;
```
Comment (aheejin) [inline test/bad_params.asm.js]: This should NOT trigger clang-format error

File: src/wasm.h [hunk 2]
```diff
@@ -44,6 +44,8 @@ typedef uint32_t Index;
 
 // An address in linear memory. For now only wasm32
 struct Address {
+  int
+      testvar;
```
Comment (aheejin) [inline src/wasm.h]: This is for test; will not be included in the final commit

**Comment by kripken** [comment]
Thanks! Mostly looks good, and I like how nice and clear the example error on travis is. Some questions:

 * How long does it take to run this?
 * Can we tell it to just work on `src/` perhaps?
 * Oddly the appveyor build fails on a static assert. I don't immediately see why - seems surprising?

**Comment by aheejin** [comment]
> Thanks! Mostly looks good, and I like how nice and clear the example error on travis is. Some questions:
> 
> * How long does it take to run this?

A fraction of second, maybe.

> * Can we tell it to just work on `src/` perhaps?

Done.

> * Oddly the appveyor build fails on a static assert. I don't immediately see why - seems surprising?

Don't know.. I don't think it's related though. My next commit will retrigger the Travis CI and we'll see.

**Comment by aheejin** [comment]
I got offline suggestion from @kripken that we maybe can run this on the whole codebase + also on a separate job, but can we land this first and improve later, because #2048 and other PRs are stalled on this?

**Comment by aheejin** [comment]
@tlively Would this clang-format on/off header/footer in gen-s-parser.py be ok?

**Pull request merged**